tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -2652,4 +2657,54 @@ public function userHasStoredEtagOfElementOnPathFromSpace($user, $path, $storePa
 			throw new Exception("Expected stored etag to be some string but found null!");
 		}
 	}
+
+	/**
+	 * @When /^user "([^"]*)" creates public link share of the space "([^"]*)" with settings:$/
+	 *
+	 * @param  string $user
+	 * @param  string $spaceName
+	 * @param TableNode|null $table
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function sendShareSpaceViaLinkRequest(
+		string $user,
+		string $spaceName,
+		?TableNode $table
+	): void {
+		$space = $this->getSpaceByName($user, $spaceName);
+		$rows = $table->getRowsHash();
+
+		$rows["shareType"] = \array_key_exists("shareType", $rows) ? $rows["shareType"] : 3;
+		$rows["permissions"] = \array_key_exists("permissions", $rows) ? $rows["permissions"] : null;
+		$rows["password"] = \array_key_exists("password", $rows) ? $rows["password"] : null;
+		$rows["name"] = \array_key_exists("name", $rows) ? $rows["name"] : null;
+		$rows["expireDate"] = \array_key_exists("expireDate", $rows) ? $rows["expireDate"] : null;
+		
+		$body = [
+			"space_ref" => $space['id'],
+			"shareType" => $rows["shareType"],
+			"permissions" => $rows["permissions"],
+			"password" => $rows["password"],
+			"name" => $rows["name"],
+			"expireDate" => $rows["expireDate"]
+		];
+
+		$fullUrl = $this->baseUrl . $this->ocsApiUrl;
+
+		$this->featureContext->setResponse(
+			HttpRequestHelper::post(
+				$fullUrl,
+				"",
+				$user,
+				$this->featureContext->getPasswordForUser($user),
+				[],
+				$body
+			)
+		);
```

Comment:
This function seems to be used in `SpacesContext` repeatedly. May be we can make a single function and use it every time when we use a `POST` method.